src/WinUIEx/WebAuthenticator.cs

```diff
@@ -258,9 +270,11 @@ private async Task<WebAuthenticatorResult> Authenticate(Uri authorizeUri, Uri ca
                 }
             }
 
+            var newUri = BeforeProcessStart != null ? BeforeProcessStart(authorizeUri) : authorizeUri;
```

Comment:
Passing `state` here as well might make the consuming code a lot easier, but wasn't sure it made sense from the perspective of the hook itself.